src/Listener/JsonApiListener.php

```diff
@@ -483,6 +483,12 @@ protected function _getEntityList($entityName, $associations)
      */
     protected function _checkRequestData()
     {
+        // prevent false positives for GET requests using query parameters
+        // as those will also have request data
```

Comment:
Could also be the `friendsofcake/search` PrgComponent perhaps? As that has this behaviour.